test/e2e/operator.go

```diff
@@ -272,3 +277,102 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var location, resourceGroup, subnet, vnet string
+	var correctNSG *string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+	type subnetDescriptor struct {
+		resourceGroup string
+		vnetName      string
+		subnetName    string
+		location      string
+	}
+	enableReconcileNSG := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileNSGs {
+			instance.Spec.Features.ReconcileNSGs = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+	getDescriptorFromProviderSpec := func(providerSpec *runtime.RawExtension) (*string, *subnetDescriptor, error) {
+		var spec azureproviderv1beta1.AzureMachineProviderSpec
+		err := json.Unmarshal(providerSpec.Raw, &spec)
+		if err != nil {
+			return nil, nil, err
+		}
+		return &spec.ResourceGroup, &subnetDescriptor{
+			resourceGroup: spec.NetworkResourceGroup,
+			vnetName:      spec.Vnet,
+			subnetName:    spec.Subnet,
+			location:      spec.Location,
+		}, nil
+	}
+	gatherNetworkInfo := func() {
+		machines, err := clients.MachineAPI.MachineV1beta1().Machines("openshift-machine-api").List(ctx, metav1.ListOptions{LabelSelector: "machine.openshift.io/cluster-api-machine-role=worker"})
+		Expect(err).NotTo(HaveOccurred())
+		subnetMap := make(map[subnetDescriptor]bool)
+		for _, machine := range machines.Items {
+			_, subnetDesc, err := getDescriptorFromProviderSpec(machine.Spec.ProviderSpec.Value)
+			if err != nil {
+				log.Fatal(err)
+			}
+			subnetMap[*subnetDesc] = true
+		}
+		for ws := range subnetMap {
```

Comment:
Q: Is the length of `subnetMap` always equal to 1?

Reply:
I have gotten rid of this bit of code so dont use this loop over subnetmap anymore